libsolidity/lsp/ReferenceCollector.h

```diff
@@ -0,0 +1,69 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <libsolidity/ast/ASTForward.h>
+#include <libsolidity/ast/ASTVisitor.h>
+
+namespace solidity::lsp
+{
+
+enum class DocumentHighlightKind
+{
+	Unspecified,    //!< could be for example a highlight found in a comment
+	Text,           //!< a textual occurrence
+	Read,           //!< read access to a variable
+	Write,          //!< write access to a variable
+};
+
+// Represents a symbol / AST node that is to be highlighted, with some context associated.
+using Reference = std::tuple<langutil::SourceLocation, DocumentHighlightKind>;
```

Comment:
Are we possibly expecting more context information in the future? If so, would it make more sense to make the second parameter a struct that contains `highightKind` or so?